composition-js/src/__tests__/compose.test.ts

```diff
@@ -4630,4 +4642,75 @@ describe('composition', () => {
     const authenticatedDirectiveExists = schema.directives().find(d => d.name === 'authenticated');
     expect(authenticatedDirectiveExists).toBeUndefined();
   });
+
+  it('@source{API,Type,Field} directives', () => {
+    const subgraphA = {
+      typeDefs: gql`
```

Comment:
Add some more tests for some of the special cases? i.e. multiple graphs with the same args, etc.

Reply:
https://github.com/apollographql/federation/pull/2894/commits/59fd58f994a1cf106bf9bd478f0b9bb511e7f68a